### PR TITLE
Remove circular dependencies

### LINKED ISSUE #718: Circular dependency between Store and Emitter
**What happened**:
We have a circular dependency between Store and Emitter in the frontend.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know**:

**Environment**:

### PR DESCRIPTION
**Pull request by holgerkoser**
**What this PR does / why we need it**:
This PR removes the circular dependency between `store` and `emitter` via a Vuex plugin and between `store` and `utils`. I addition it ensures that the store is only used correctly via the component API and not via direct import. I also added the [circular-dependency-plugin](https://github.com/aackerman/circular-dependency-plugin) to the webpack config.

**Which issue(s) this PR fixes**:
Fixes #718

**Special notes for your reviewer**:

**Release note**:

```bugfix user

```

### CODE REVIEW

File: frontend/src/store/index.js [hunk 1]
```diff
@@ -1010,6 +1020,198 @@ const getters = {
   },
   onlyShootsWithIssues (state, getters) {
     return getters['shoots/onlyShootsWithIssues']
+  },
+  projectNameByNamespace (state, getters) {
+    return ({ namespace } = {}) => {
+      const project = find(getters.projectList, ['metadata.namespace', namespace])
+      return get(project, 'metadata.name') || replace(namespace, /^garden-/, '')
```
Comment (petersutter) [inline frontend/src/store/index.js:1027]: Do we still need this fallback?

Comment (holgerkoser) [reply frontend/src/store/index.js:1027]: I don't know. But I don't wanted to start changing any code in this PR.

Comment (petersutter) [reply frontend/src/store/index.js:1027]: ok

**Review by grolu — APPROVED** [review]
/lgtm

**Pull request merged**